providers/os/resources/asset_eol.go

```diff
@@ -67,7 +67,17 @@ func initAssetEol(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[st
 	// get new advisory report
 	// start scanner client
 
-	scannerClient, err := newAdvisoryScannerHttpClient(mcc.ApiEndpoint, mcc.Plugins, mcc.HttpClient)
```

Comment:
This didn't work as expected before:
```
admin@ip-10-0-101-143:~$ ./cnquery version
cnquery 8.28.4 (f7739318, 2023-09-20T10:09:07Z)
admin@ip-10-0-101-143:~$ ./cnquery run --config sa-edge-default.json -c "platform.eol"
→ no provider specified, defaulting to local.
  Use --help for a list of available providers.
→ loaded configuration from sa-edge-default.json using source --config
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
Query encountered errors:
failed to create resource 'platform.eol': failed to do request: Post "https://api.edge.mondoo.com/AdvisoryScanner/IsEol": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
platform.eol: no data available
```

It would bypass the proxy and fetch data from upstream. The actual HTTP client was working, but it ignored the proxy.

Reply:
Thank you for verifying!

Reply:
I ran a test myself with a local proxy, because this looked like it should have worked with the initial version. I got it working and verified on the proxy as well:

```
CONNECT   Oct 20 23:57:35.271 [1928765]: Connect (file descriptor 4): 127.0.0.1
CONNECT   Oct 20 23:57:35.271 [1928765]: Request (file descriptor 4): CONNECT us.api.mondoo.com:443 HTTP/1.1
INFO      Oct 20 23:57:35.271 [1928765]: No upstream proxy for us.api.mondoo.com
INFO      Oct 20 23:57:35.271 [1928765]: opensock: opening connection to us.api.mondoo.com:443
INFO      Oct 20 23:57:35.293 [1928765]: opensock: getaddrinfo returned for us.api.mondoo.com:443
CONNECT   Oct 20 23:57:35.346 [1928765]: Established connection to host "us.api.mondoo.com" using file descriptor 5.
INFO      Oct 20 23:57:35.346 [1928765]: Not sending client headers to remote machine
INFO      Oct 20 23:57:35.583 [1928765]: Closed connection between local client (fd:4) and remote client (fd:5)
```

```bash
> cnquery run docker image 216c552ea5ba --config ~/.config/mondoo/mondoo.yml -c "asset.eol"
! using builtin provider for os
! using builtin provider for os
→ loaded configuration from /home/zero/.config/mondoo/mondoo.yml using source --config
asset.eol: asset.eol date=2027-03-31 17:00:00 -0700 PDT
```

That you for checking tho, i suspect it may have been missing the provider update before? I used the builtin os provider to make sure it used this very code-path.